File: source/WeddingManager.Web/js/services/summaryService.js
(function(){
    angular.module('weddingManager')
    .service('summaryService', function(summaryFactory){
        var service = {};
        var _companySummary = null;
        
        service.refreshCompanySummary = function(companyId, onSuccess, onError){
            summaryFactory.retrieveCompanySummary(companyId, function(companySummary){
                _companySummary = companySummary;
                onSuccess(_companySummary);
            }, onError);
        }
        
        return service;
    });
})();

File: source/WeddingManager.Services/SummaryService.cs
using WeddingManager.Core.Data;
using WeddingManager.Core.Repositories;
using WeddingManager.Core.Services;

namespace WeddingManager.Services
{
    public class SummaryService : ISummaryService
    {
        public SummaryService(ISummaryRepository summaryRepository)
        {
            _summaryRepository = summaryRepository;
        }

        public CompanySummary RetrieveCompanySummary(int companyId)
        {
            return _summaryRepository.RetrieveCompanySummary(companyId);
        }

        public ExpensesSummary RetrieveExpensesSummary(int companyId)
        {
            return _summaryRepository.RetrieveExpensesSummary(companyId);
        }

        private readonly ISummaryRepository _summaryRepository;
    }
}